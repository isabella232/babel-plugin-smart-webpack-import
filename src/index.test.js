/* eslint-disable no-template-curly-in-string */
import createBabylonOptions from "babylon-options"
import pluginTester from "babel-plugin-tester"

import thisPlugin from "./"

const babelOptions = {
  filename: "current/folder/currentFile.js",
  parserOpts: createBabylonOptions({
    sourceType: "module",
    plugins: [ "dynamicImport" ]
  })
}

pluginTester({
  plugin: thisPlugin,
  babelOptions,
  snapshot: true,
  tests: {
    "Adds chunkname when missing": "import('./HelloView')",
    "Adds chunkname when missing for subfolder import": "import('./sub/HelloView')",
    "Supports multi targets": "import(`./views/${name}`)",
    "Simplifies deep multi targets": "import(`../other/views/${name}`)",
    "Simplifies deep multi targets with extension":
      "import(`../other/views/${name}.css`)",
    "Simplifies deep multi targets with prefix":
      "import(`../other/views/prefix-${name}.css`)",
    "Keeps chunkName simple even for longer paths":
      "import('./views/admin/SettingsView')",
    "Keeps existing comments": "import(/* keep me */ './HelloView')",
    "Adds chunkName to existing magic comment":
      "import(/* webpackPrefetch: true */ './HelloView')",
    "Don't overwrites manual chunkName":
      "import(/* webpackChunkName: 'CustomHello' */ './HelloView')",
    "Adds chunkname when missing inside loadable()": "loadable((props) => import('./HelloView'))"
  }
})
