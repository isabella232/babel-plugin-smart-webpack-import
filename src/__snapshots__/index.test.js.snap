// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smart-webpack-import Adds chunkName to existing magic comment: Adds chunkName to existing magic comment 1`] = `
"
import(/* webpackPrefetch: true */ './HelloView')

      ↓ ↓ ↓ ↓ ↓ ↓

import(
/*webpackPrefetch:true,webpackChunkName:'HelloView-PzV81'*/

/**/
'./HelloView');
"
`;

exports[`smart-webpack-import Adds chunkname when missing: Adds chunkname when missing 1`] = `
"
import('./HelloView')

      ↓ ↓ ↓ ↓ ↓ ↓

import(
/*webpackChunkName:'HelloView-PzV81'*/
'./HelloView');
"
`;

exports[`smart-webpack-import Don't overwrites manual chunkName: Don't overwrites manual chunkName 1`] = `
"
import(/* webpackChunkName: 'CustomHello' */ './HelloView')

      ↓ ↓ ↓ ↓ ↓ ↓

import(
/* webpackChunkName: 'CustomHello' */
'./HelloView');
"
`;

exports[`smart-webpack-import Keeps chunkName simple even for longer paths: Keeps chunkName simple even for longer paths 1`] = `
"
import('./views/admin/SettingsView')

      ↓ ↓ ↓ ↓ ↓ ↓

import(
/*webpackChunkName:'SettingsView-hOkkh'*/
'./views/admin/SettingsView');
"
`;

exports[`smart-webpack-import Keeps existing comments: Keeps existing comments 1`] = `
"
import(/* keep me */ './HelloView')

      ↓ ↓ ↓ ↓ ↓ ↓

import(
/*webpackChunkName:'HelloView-PzV81'*/

/* keep me */
'./HelloView');
"
`;
